### PR TITLE
UI-3252-fixes: Fix create port button and return required data on port edit

### PR DESCRIPTION
**Pull request by masmerino13**

### CODE REVIEW

File: src/apps/common/submodules/portListing/portListing.js [hunk 1]
```diff
@@ -780,108 +867,115 @@ define(function(require) {
...
+								needFix: isLastAndRejected(transition, idx, array),
+								status: transition.transition.new,
+								timestamp: moment(monster.util.gregorianToDate(transition.timestamp)).valueOf(),
+								// only show the "Action Required" text if the last transition's status is "rejected"
+								title: self.i18n.active().portListing.misc[isLastAndRejected(transition, idx, array) ? 'alternateStatus' : 'status'][transition.transition.new]
+							};
+						})
+						.value()
+				])
+				.reduce(function(acc, item) {
```
Comment (guillegr123) [inline src/apps/common/submodules/portListing/portListing.js]: I think this `reduce(...)` could be replaced by [`flatten()`](https://lodash.com/docs/4.17.11#flatten)

File: src/apps/common/submodules/portListing/portListing.js [hunk 2]
```diff
@@ -894,29 +988,13 @@ define(function(require) {
...
-										},
-										id: request.id,
-										name: request.name,
-										reference: request.carrier_reference_number,
-										state: request.port_state
-									};
-								});
+								return account.port_requests;
 							})
 							.reduce(function(acc, item) {
```
Comment (guillegr123) [inline src/apps/common/submodules/portListing/portListing.js]: This `reduce(...)` could be replaced by `flatten()` too.

File: src/apps/common/submodules/portListing/portListing.js [hunk 3]
```diff
@@ -926,25 +1004,72 @@ define(function(require) {
 					.reduce(function(acc, item) {
```
Comment (guillegr123) [inline src/apps/common/submodules/portListing/portListing.js]: This other `reduce(...)` could be replaced too with `flatten()`.

File: src/apps/common/submodules/portListing/portListing.js [hunk 4]
```diff
@@ -926,25 +1004,72 @@ define(function(require) {
...
+					.chain(results)
+					.map(function(payload) {
+						return _.map(payload, function(account) {
+							return {
+								id: account.account_id,
+								name: account.account_name
+							};
+						});
+					})
+					.reduce(function(acc, item) {
```
Comment (guillegr123) [inline src/apps/common/submodules/portListing/portListing.js]: And maybe this other `reduce` too :sweat_smile:

File: src/apps/common/submodules/portListing/portListing.js [hunk 5]
```diff
@@ -30,41 +35,45 @@ define(function(require) {
...
 			if (self.appFlags.portListing.isMonsterApp) {
-				self.portListingRenderLayout(args);
+				self.appFlags.portListing.parent = args.parent;
+				self.appFlags.portListing.container = args.container;
 			} else {
-				if (args.parent && args.parent.getId()) {
-					modal = args.parent;
-				} else {
-					modal = monster.ui.fullScreenModal(null, {
+				self.appFlags.portListing.parent = _.has(args, 'parent.getId')
```
[the remark this answers is not in the document]
Comment (azefiel) [reply src/apps/common/submodules/portListing/portListing.js]: True, good catch!

The goal of this ternary though, is to determine if `args.parent` is an already instantiated fullScreenModal (which has a `getId` [method](https://github.com/2600hz/monster-ui/blob/master/src/js/lib/monster.ui.js#L2969) by opposition to a regular jQuery object) so the app can know if it need to create a new fullScreenModal or not.

File: src/apps/common/submodules/portListing/portListing.js [hunk 6]
```diff
@@ -74,58 +83,62 @@ define(function(require) {
...
-					}));
-
-					self.portListingBindListingEvents(template, _.merge({}, args, {
-						data: {
-							ports: _.keyBy(ports, 'id')
-						}
-					}));
+					self.portListingRenderListingIncomplete({
+						template: template,
+						portRequests: portRequests
```
[the remark this answers is not in the document]
Comment (azefiel) [reply src/apps/common/submodules/portListing/portListing.js]: Agreed, this was improved upon when merged to `master`.

File: src/apps/common/submodules/portListing/portListing.js (hunk 1 above)
Comment (azefiel) [reply src/apps/common/submodules/portListing/portListing.js]: Nice, I will improve this in my next PR (UI-3259).

**Pull request merged**